stacks-signer/src/cli.rs

```diff
@@ -170,11 +174,83 @@ pub struct GenerateFilesArgs {
     pub timeout: Option<u64>,
 }
 
+#[derive(Clone, Debug)]
+/// Wrapper around `Pox4SignatureTopic` to implement `ValueEnum`
+pub struct StackingSignatureMethod(Pox4SignatureTopic);
+
+impl StackingSignatureMethod {
+    /// Get the inner `Pox4SignatureTopic`
+    pub fn topic(&self) -> &Pox4SignatureTopic {
+        &self.0
+    }
+}
+
+impl From<Pox4SignatureTopic> for StackingSignatureMethod {
+    fn from(topic: Pox4SignatureTopic) -> Self {
+        StackingSignatureMethod(topic)
+    }
+}
+
+impl ValueEnum for StackingSignatureMethod {
+    fn to_possible_value(&self) -> Option<clap::builder::PossibleValue> {
+        Some(clap::builder::PossibleValue::new(self.0.as_str()))
+    }
+
+    fn value_variants<'a>() -> &'a [Self] {
+        &[
+            StackingSignatureMethod(Pox4SignatureTopic::StackStx),
+            StackingSignatureMethod(Pox4SignatureTopic::StackExtend),
+            StackingSignatureMethod(Pox4SignatureTopic::AggregationCommit),
+        ]
+    }
+
+    fn from_str(input: &str, _ignore_case: bool) -> Result<Self, String> {
+        let topic = match input {
+            "stack-stx" => Pox4SignatureTopic::StackStx,
+            "stack-extend" => Pox4SignatureTopic::StackExtend,
+            "aggregation-commit" => Pox4SignatureTopic::AggregationCommit,
+            "agg-commit" => Pox4SignatureTopic::AggregationCommit,
+            _ => return Err(format!("Invalid topic: {}", input)),
+        };
+        Ok(topic.into())
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+/// Arguments for the generate-stacking-signature command
+pub struct GenerateStackingSignatureArgs {
```

Comment:
It would be useful to have test vectors with hex strings, e.g. containing the serialized payload for this struct. That way other folks could leverage the same vectors to be more confident in in their implementation (like in a js one). 

Not super important though unless we actually envision needing to implement similar code in js any time soon.